core/test/nuxeo-task-page-provider.test.js

```diff
@@ -0,0 +1,204 @@
+        expect(request.queryParams).to.have.property('pageSize', 40);
+        expect(request.queryParams).to.have.property('currentPageIndex', 0);
+      }
+
+      setup(() => {
+        server.respondWith('get', '/api/v1/task', {
+          entries: [{}],
+          numberOfPages: 1,
+          resultsCount: 1,
+          isNextPageAvailable: false,
+          currentPageOffset: 0,
+          pageSize: 10,
+          isPreviousPageAvailable: false,
+          currentPageSize: 1,
+        });
+      });
+
+      test("Should not include parameters' entries when they are null or undefined", async () => {
+        const params = {
+          myNull: null,
+          myUndefined: null,
+          myNonNull: 'test',
+        };
+
+        const pageProvider = await fixture(html`
+          <nuxeo-task-page-provider params="${JSON.stringify(params)}"></nuxeo-task-page-provider>
+        `);
+        await pageProvider.fetch();
+
+        const request = server.getLastRequest('get', '/api/v1/task');
+        assertBaseRequest(request);
+        expect(request.queryParams).to.have.property('myNonNull', 'test');
+        expect(request.queryParams).to.not.have.any.keys('myNull', 'myUndefined');
+      });
+
+      test('Should contain mandatory parameters when no parameters are passed as argument', async () => {
+        const pageProvider = await fixture(html`
+          <nuxeo-task-page-provider offset="5"></nuxeo-task-page-provider>
+        `);
+        await pageProvider.fetch();
+
+        const request = server.getLastRequest('get', '/api/v1/task');
+        assertBaseRequest(request);
+        expect(request.queryParams).to.have.property('offset', 5);
+      });
+
+      test('Should convert non-default parameters to string when they have a different type', async () => {
+        const params = {
+          myNumber: 1,
+          myArray: ['string 1', 'string 2'],
+          myUnknownObject: {
+            key: 'value',
+            otherKey: 123,
+          },
+          myBoolean: false,
+        };
+        const pageProvider = await fixture(html`
+          <nuxeo-task-page-provider params="${JSON.stringify(params)}"></nuxeo-task-page-provider>
+        `);
+        await pageProvider.fetch();
+
+        const request = server.getLastRequest('get', '/api/v1/task');
+        assertBaseRequest(request);
+        expect(request.queryParams.myNumber).to.be.a('string');
+        expect(request.queryParams.myNumber).to.be.equal('1');
+
+        expect(request.queryParams.myArray).to.be.a('string');
+        expect(request.queryParams.myArray).to.be.equal('["string 1","string 2"]');
+
+        expect(request.queryParams.myUnknownObject).to.be.a('string');
+        expect(request.queryParams.myUnknownObject).to.be.equal('{"key":"value","otherKey":123}');
+
+        expect(request.queryParams.myBoolean).to.be.a('string');
+        expect(request.queryParams.myBoolean).to.be.equal('false');
```

Comment:
should we check the type, as for each expectation you check the equality and i think (i talk under you control) the expect will fail if there is not equality (my main question `to.be.equal`  will it do the equivalent of `===` )

Reply:
Indeed, copy/paste error. 😄 
The documentation for this specific method: https://www.chaijs.com/api/bdd/#method_equal